connection/kubernetes.go

```diff
@@ -51,6 +51,33 @@ type KubernetesConnection struct {
 	EKS  *EKSConnection  `json:"eks,omitempty"`
 	GKE  *GKEConnection  `json:"gke,omitempty"`
 	CNRM *CNRMConnection `json:"cnrm,omitempty"`
+
+	// For tests and special cases
+	CustomClientSet  kubernetes.Interface
+	CustomRestConfig *rest.Config
+}
+
+func (c KubernetesConnection) Hash() string {
+	if c.ConnectionName != "" {
+		return "connection=" + c.ConnectionName
+	}
+	if c.Kubeconfig != nil {
+		return "kubeconfig=" + c.Kubeconfig.String()
+	}
+	if c.EKS != nil {
+		return fmt.Sprintf("eks=%s/%v", c.EKS.Cluster, c.EKS.ToModel())
+	}
+	if c.GKE != nil {
+		return fmt.Sprintf("gke=%s/%v", c.GKE.Cluster, c.GKE.ToModel())
+	}
+	if c.CNRM != nil {
+		return fmt.Sprintf("cnrm=%s/%v", c.CNRM.ClusterResource, c.GKE.ToModel())
+	}
+	return "local"
+}
+
+func (c KubernetesConnection) CanExpire() bool {
+	return c.EKS != nil || c.GKE != nil || c.CNRM != nil
```

Comment:
It can also expire if reading kubeconfig from a secret

Reply:
Updated